src/dune_rules/pkg_build_progress.ml

```diff
@@ -0,0 +1,55 @@
+open! Import
+
+let enabled = Config.make_toggle ~name:"pkg_build_progress" ~default:`Disabled
+
+let display package_name package_version status =
+  let status_tag = User_message.Style.Ok in
+  let status_text =
+    match status with
+    | `Downloading -> "Downloading"
+    | `Building -> "Building"
+  in
+  let user_message =
+    User_message.make
+      [ Pp.concat
+          [ Pp.tag status_tag (Pp.text status_text)
+          ; Pp.textf
+              " %s.%s"
+              (Package.Name.to_string package_name)
+              (Package_version.to_string package_version)
+          ]
+      ]
+  in
+  Console.print_user_message user_message
+;;
+
+let display_build_progress package_name package_version ~source_dir ~target_dir =
+  let open Action_builder.O in
+  let source =
+    match source_dir with
+    | `No_package_source -> Action_builder.return ()
```

Comment:
This might need to be extended because while we don't download source tarballs in some packages (like the classic `seq.base`) we still download extra-sources so I would expect them to show up in as "Downloading".

Reply:
Fixed